### PR TITLE
Create API for `.ar*`, `aei`, `ar <reg>=<val>`

### PR DESCRIPTION
**Pull request by ret2libc**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

* Create API for `.ar*`, `aei`, `ar <reg>=<val>`
* Move rz_debug_reg_list from RzDebug to RzCore and directly use rz_cons_printf

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

File: librz/core/cdebug.c [hunk 1]
```diff
@@ -19,9 +20,222 @@ RZ_API bool rz_core_debug_step_one(RzCore *core, int times) {
...
+				pj_kn(pj, item->name, value);
+			} else {
+				if (pr && pr->wide_offsets && dbg->bits & RZ_SYS_BITS_64) {
+					snprintf(strvalue, sizeof(strvalue), "0x%016" PFMT64x, value);
+				} else {
+					snprintf(strvalue, sizeof(strvalue), "0x%08" PFMT64x, value);
+				}
+			}
+		} else {
+			value = rz_reg_get_value_big(dbg->reg, item, &valueBig);
```
Comment (wargio) [inline librz/core/cdebug.c]: value is never used after this assignment.

File: librz/core/agraph.c [hunk 2]
```diff
@@ -5,6 +5,7 @@
 #include <rz_util/rz_graph_drawable.h>
 #include <ctype.h>
 #include <limits.h>
+#include "private.h"
```
Comment (wargio) [inline librz/core/agraph.c]: This header name feels wrong.

**Review by wargio — CHANGES_REQUESTED** [review]

File: librz/core/cdebug.c (hunk 1 above)
Comment (ret2libc) [reply librz/core/cdebug.c]: this function is all copy-pasted code moved from rz_debug_reg_List :(

Comment (ret2libc) [reply librz/core/cdebug.c]: didn't even look at it so far

File: librz/core/agraph.c (hunk 2 above)
Comment (ret2libc) [reply librz/core/agraph.c]: suggestions? It is used also in other modules so far e.g. io_private.h , bin/i/private.h

File: librz/core/cdebug.c (hunk 1 above)
Comment (wargio) [reply librz/core/cdebug.c]: i guess you can just delete the line.

File: librz/core/agraph.c (hunk 2 above)
Comment (wargio) [reply librz/core/agraph.c]: `io_private` seems much better

**Review by wargio — APPROVED** [review]

**Review by XVilka — CHANGES_REQUESTED** [review]
I see almost everywhere `rz_core_debug_reg_list()` called with `core->dbg` as the first parameter. What about using `RzCore *core` then instead of `RzDebug *dbg`? It will make using corebind unnecessary.

**Comment by ret2libc** [comment]
> I see almost everywhere `rz_core_debug_reg_list()` called with `core->dbg` as the first parameter. What about using `RzCore *core` then instead of `RzDebug *dbg`? It will make using corebind unnecessary.

Good point, done.

File: librz/core/cmd_analysis.c [hunk 3]
```diff
@@ -6107,31 +6142,8 @@ static void cmd_analysis_esil(RzCore *core, const char *input) {
 			rz_analysis_esil_free(esil);
 			core->analysis->esil = NULL;
 			break;
-		case 0: //lolololol
```
Comment (XVilka) [inline librz/core/cmd_analysis.c:6110]: lolololol

**Review by XVilka — APPROVED** [review]

**Pull request merged**